src/sentry/sentry_metrics/indexer/postgres_v2.py

```diff
@@ -90,6 +91,20 @@ class KeyResults:
     def __init__(self) -> None:
         self.results: MutableMapping[int, MutableMapping[str, int]] = defaultdict(dict)
 
+    @classmethod
+    def merge_results(
+        cls,
+        result_mappings: Sequence[Mapping[int, Mapping[str, int]]],
+    ) -> Mapping[int, Mapping[str, int]]:
+        new_results: MutableMapping[int, MutableMapping[str, int]] = defaultdict(dict)
+        for result_map in result_mappings:
+            for org_id, strings in result_map.items():
+                new_results[org_id].update(strings)
+        return new_results
+
+    def add_key_result(self, key_result: KeyResult) -> None:
+        self.results[key_result.org_id].update({key_result.string: key_result.id})
```

Comment:
I think this does not work.
What if the org id is not yet present in `self.results` ?

Reply:
`self.results` is a `defaultdict(dict)` so it should just be updating an empty dict iirc